ignite/pkg/jsonfile/jsonfile_test.go

```diff
@@ -0,0 +1,310 @@
+				WithKeyValue(
+					"consensus_params.block.max_bytes",
+					"3000000",
+				),
+				WithKeyIntValue(
+					"consensus_params.block.time_iota_ms",
+					1000,
+				),
+				WithTime(
+					"genesis_time",
+					999999999,
+				),
+			},
+		},
+		{
+			name:     "add non-existing field",
+			filepath: "testdata/jsonfile.json",
+			opts: []UpdateFileOption{
+				WithKeyValue(
+					"app_state.auth.params.sig_verify_cost_ed25519",
+					"111",
+				),
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			f, err := FromPath(tt.filepath)
+			require.NoError(t, err)
+			err = f.Update(tt.opts...)
+			if tt.err != nil {
+				require.Error(t, err)
+				require.ErrorIs(t, err, tt.err)
+				return
+			}
+			require.NoError(t, err)
+
+			updates := make(map[string][]byte)
+			for _, opt := range tt.opts {
+				opt(updates)
+			}
+			for key, value := range updates {
+				var out interface{}
+				err = f.Field(key, &out)
+				switch out := out.(type) {
+				case string:
+					require.Equal(t, strings.ReplaceAll(string(value), "\"", ""), out)
+				case float64:
+					v, err := strconv.Atoi(string(value))
+					require.NoError(t, err)
+					require.Equal(t, v, int(out))
+				default:
+					require.Equal(t, value, out)
+				}
+			}
+		})
+	}
+}
+
+func TestJSONFile_Hash(t *testing.T) {
+	tests := []struct {
+		name     string
+		filepath string
+		want     string
+		err      error
+	}{
+		{
+			name:     "file hash",
+			filepath: "testdata/jsonfile.json",
+			want:     "4d685d9cb6f9fb9815a33f10a75cd9970f162bbc6ebc8c5c0e3fd166d1b3ee93",
+		},
+		{
+			name:     "not found file",
+			filepath: "testdata/genesis_not_found.json",
+			err: errors.New(
+				"cannot open the file: open testdata/genesis_not_found.json: no such file or directory",
+			),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			f, err := FromPath(tt.filepath)
+			if tt.err != nil {
+				require.Error(t, err)
+				require.Equal(t, err.Error(), tt.err.Error())
+				return
+			}
+			require.NoError(t, err)
+			got, err := f.Hash()
+			if tt.err != nil {
+				require.Error(t, err)
+				require.ErrorIs(t, err, tt.err)
+				return
+			}
+			require.NoError(t, err)
+			require.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func TestFromURL(t *testing.T) {
+	type args struct {
+		url             string
+		tarballFileName string
+	}
+	tests := []struct {
+		name        string
+		args        args
+		verifyField string
+		wantField   string
+		err         error
+	}{
+		{
+			name: "JSON URL",
+			args: args{
+				url: "https://raw.githubusercontent.com/ignite-hq/cli/4d4f6b436d15aa3fb8aeca4bf91b6a557f897f9b/ignite/pkg/tarball/testdata/example.json",
+			},
+			verifyField: "chain_id",
+			wantField:   "gaia-1",
+		},
+		{
+			name: "tarball URL",
+			args: args{
+				url:             "https://github.com/ignite-hq/cli/raw/4d4f6b436d15aa3fb8aeca4bf91b6a557f897f9b/ignite/pkg/tarball/testdata/example-subfolder.tar.gz",
```

Comment:
We could use httptest.Server in order to avoid real http requests inside of the tests